### PR TITLE
refactor(Reports): change component class for function

### PR DESCRIPTION
**Pull request by fcoronelmakingsense**

### CODE REVIEW

File: src/components/Reports/Reports.js [hunk 1]
```diff
@@ -15,54 +15,47 @@ import { Helmet } from 'react-helmet';
 import { Loading } from '../Loading/Loading';
 import { addDays } from '../../utils';
 
-class Reports extends React.Component {
-  /**
-   * @param { Object } props
-   * @param { import('../../services/pure-di').AppServices } props.dependencies
```
Comment (andresmoschini) [inline src/components/Reports/Reports.js]: Maybe you would like to keep these annotations, they are useful for IntelliSense.

There should be already examples of function components with these annotations.

File: src/components/Reports/Reports.js [hunk 2]
```diff
@@ -15,54 +15,47 @@ import { Helmet } from 'react-helmet';
...
-   */
-  constructor({ dependencies: { datahubClient, appConfiguration } }) {
-    super();
+const periodSelectedDaysDefault = 7;
 
-    this.datahubClient = datahubClient;
-    this.appConfiguration = appConfiguration;
```
Comment (andresmoschini) [inline src/components/Reports/Reports.js]: ❤️

File: src/components/Reports/Reports.js (hunk 1 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/Reports.js]: ohh, thanks didn't know!

File: src/components/Reports/Reports.js [hunk 3]
```diff
@@ -15,54 +15,47 @@ import { Helmet } from 'react-helmet';
...
-    const now = new Date();
-    const dateFrom = addDays(now, days * -1);
-    this.setState({ periodSelectedDays: days, dateFrom: dateFrom, dateTo: now });
-  };
+  useEffect(() => {
+    const fetchData = async () => {
+      const domains = await datahubClient.getAccountDomains();
+      if (domains) {
+        const now = new Date();
+        const dateFrom = addDays(now, periodSelectedDaysDefault * -1);
```
Comment (andresmoschini) [inline src/components/Reports/Reports.js]: What do you think about adding _periodSelectedDays_ as a dependency and update it declaratively?

**Comment by cbernat** [comment]
💯

**Comment by andresmoschini** [comment]
❤️

**Pull request merged**